### ISSUE TITLE
expect().toHaveProperty() throw segfault if expect value is undefined

### ISSUE THREAD

**Issue by vesamet** [oven-sh/bun#3135]
### What version of Bun is running?

0.6.5

### What platform is your computer?

Microsoft Windows NT 10.0.19044.0 x64

### What steps can reproduce the bug?

Run the follow test:

```js
import { expect, test } from "bun:test";
test("toHaveProperty() returns Segfault", () => {
  expect(undefined).toHaveProperty("anyProperty");
});

```

### What is the expected behavior?

Should return an error like 
```
error: expect(received).toHaveProperty(path)

Expected path: "anyProperty"

Unable to find property

```

### What do you see instead?

``` 

Segmentation fault

```

**Comment by v3nntrnx** [comment]
I'll look into this

**Comment by Electroid** [comment]
Thanks for reporting! A fix be released when #3071 / 878b90caefcd4a65a537e0fc06ca9f86d3f5a39a is merged and 0.6.6 is released later today.
